### PR TITLE
SQL: Respect default timezone for TIME_PARSE and TIME_SHIFT.

### PR DESCRIPTION
**Pull request by gianm**
They were inadvertently using UTC rather than the default timezone.
Also, harmonize how time functions handle their parameters.

### CODE REVIEW

**Comment by gianm** [comment]
I don't think there's a reason to backport this to 0.15.0 @fjy

**Comment by gianm** [comment]
Getting a weird error from TC when running the inspections:

> Warning: No enabled compatible agents for this build configuration. Please register a build agent or tweak build configuration requirements.

I'll try closing and reopening.

File: sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java [hunk 1]
```diff
@@ -5995,7 +5998,7 @@ public void testTimeseriesUsingTimeFloorWithTimeShift() throws Exception
                         .setVirtualColumns(
                             expressionVirtualColumn(
                                 "v0",
-                                "timestamp_floor(timestamp_shift(\"__time\",'P1D',-1),'P1M',null,'UTC')",
+                                "timestamp_floor(timestamp_shift(\"__time\",'P1D',-1,'UTC'),'P1M',null,'UTC')",
```
Comment (jihoonson) [inline sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java]: Should we add a test with non UTC timezone as well?

Comment (gianm) [reply sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java]: I added one in ExpressionsTest.

Comment (jihoonson) [reply sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java]: Thanks.

**Comment by fjy** [comment]
Merging given that TC is broken.

**Pull request merged**